From fa80b511f54493eda0d5fb6ef17c34a30b54078a Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Mon, 6 Mar 2017 23:04:09 +0100
Subject: [PATCH 6/7] errtracker: fix testing outside of ubuntu

Unfortunately the distribution "ubuntu" is special cased so that it
shows up as "Ubuntu". Tests are not taking this into account and were
always expecting the upper-cased distribution name. This patch fixes
this.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
---
 errtracker/errtracker_test.go | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/errtracker/errtracker_test.go b/errtracker/errtracker_test.go
index cd69d30..3a433dc 100644
--- a/errtracker/errtracker_test.go
+++ b/errtracker/errtracker_test.go
@@ -84,9 +84,15 @@ func (s *ErrtrackerTestSuite) TestReport(c *C) {
 			var data map[string]string
 			err = bson.Unmarshal(b, &data)
 			c.Assert(err, IsNil)
+			var distroRelease string
+			if release.ReleaseInfo.ID == "ubuntu" {
+				distroRelease = fmt.Sprintf("%s %s", strings.Title(release.ReleaseInfo.ID), release.ReleaseInfo.VersionID)
+			} else {
+				distroRelease = fmt.Sprintf("%s %s", release.ReleaseInfo.ID, release.ReleaseInfo.VersionID)
+			}
 			c.Check(data, DeepEquals, map[string]string{
 				"ProblemType":        "Snap",
-				"DistroRelease":      fmt.Sprintf("%s %s", strings.Title(release.ReleaseInfo.ID), release.ReleaseInfo.VersionID),
+				"DistroRelease":      distroRelease,
 				"HostSnapdBuildID":   hostBuildID,
 				"CoreSnapdBuildID":   coreBuildID,
 				"SnapdVersion":       "some-snapd-version",
-- 
2.10.2

